### PR TITLE
*: update logrus and parser package, tiny clean up

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/7998
Fix https://github.com/pingcap/tidb/issues/7986
Update parser.

### What is changed and how it works?

Tiny clean up.

Tests 

 - No code

@lysu @XuHuaiyu

### CODE REVIEW

**Comment by shenli** [comment]
Please fix the CI.

**Comment by tiancaiamao** [comment]
@zz-jason PTAL

File: cmd/explaintest/run-tests.sh [hunk 1]
```diff
@@ -97,10 +94,6 @@ while getopts "t:s:r:b:v:c:i:h" opt; do
                     ;;
             esac
             ;;
-        v)
```
Comment (zz-jason) [inline cmd/explaintest/run-tests.sh]: should we also remove this flag in other "run-all-tests.sh"? for example, mysql-test, randgen-test

**Pull request merged**